<launch>

    <!--autolabor_pro_description-->
    <include file = "$(find autolabor_pro1_description)/launch/auto_pro1_laser.launch" />

    <!--autolabor pro driver-->
    <include file = "$(find autolabor_pro1_driver)/launch/driver.launch" />

    <!--key twist-->
    <node name="tele_key_twist" pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" output="screen">
        <param name="speed" value="0.2" />
        <param name="turn" value="0.3" />
    </node>
    <!--joy twist-->
    <node name="joy" pkg="joy" type="joy_node" />
    <node name = "teleop_twist_joy" pkg = "teleop_twist_joy" type = "teleop_node">
        <param name = "enable_button" value = "4" type = "int" />
        <param name = "enable_turbo_button" value = "0" />
        <param name = "scale_linear_turbo" value = "0.5" type = "double" />
        <param name = "axis_linear" value = "1" type = "int" />
        <param name = "scale_linear" value = "0.3" type = "double" />
        <param name = "axis_angular" value = "0" type = "int" />
        <param name = "scale_angular" value = "0.3" type = "double" />
    </node>

    <!--rplidar-->
    <include file = "$(find rplidar_ros)/launch/rplidar.launch" />    

    <node pkg="gmapping" type="slam_gmapping" name="gmapping">
        <param name="base_frame" value="base_footprint"/>
        <param name="odom_frame" value="odom" />
        <param name="delta" value="0.05" />
        <param name="map_update_interval" value="3.0"/>
        <param name="maxUrange" value="6.0"/>
        <param name="maxRange" value="6.0"/>
        <param name="sigma" value="0.05"/>
        <param name="kernelSize" value="1"/>
        <param name="lstep" value="0.03"/>
        <param name="astep" value="0.04"/>
        <param name="iterations" value="5"/>
        <param name="lsigma" value="0.075"/>
        <param name="ogain" value="1.0"/>
        <param name="lskip" value="0"/>
        <param name="srr" value="0.5"/>
        <param name="srt" value="0.2"/>
        <param name="str" value="0.3"/>
        <param name="stt" value="0.2"/>
        <param name="linearUpdate" value="0.01"/>
        <param name="angularUpdate" value="0.01"/>
        <param name="particles" value="6"/>
        <param name="xmin" value="-5.0"/>
        <param name="ymin" value="-5.0"/>
        <param name="xmax" value="5.0"/>
        <param name="ymax" value="5.0"/>
        <param name="minimumScore" value="60"/>
        <param name="temporalUpdate" value="3.0"/>
        <param name="resampleThreshold" value="0.5"/>
        <param name="llsamplerange" value="0.01"/>
        <param name="llsamplestep" value="0.01"/>
        <param name="lasamplerange" value="0.005"/>
        <param name="lasamplestep" value="0.005"/>
    </node>

    <!--move_base node-->
    <include file="$(find autolabor_pro1_nav)/launch/move_base_2.5.launch" />

    <!-- Run a static transform between /odom and /map -->
    <!-- <node pkg="tf" type="static_transform_publisher" name="odom_map_broadcaster" args="0 0 0 0 0 0 map odom 200" /> -->

    <!--rviz-->
    <node name="rviz" pkg="rviz" type="rviz" required="true" args="-d $(find autolabor_pro1_nav)/gmapping.rviz" />

</launch>